bindings/java/src/org/sleuthkit/datamodel/DBAccessManager.java

```diff
@@ -0,0 +1,287 @@
+		tskDB.acquireSingleUserCaseReadLock();
+
+		Statement statement = null;
+		String updateSQL = "UPDATE " + tableName + " " + sql; // NON-NLS
+
+		try {
+			statement = connection.createStatement();
+			statement.executeUpdate(updateSQL);
+		} catch (SQLException ex) {
+			throw new TskCoreException("Error Updating table " + tableName, ex);
+		} finally {
+			closeStatement(statement);
+			connection.close();
+			tskDB.releaseSingleUserCaseReadLock();
+		}
+	}
+	/**
+	 * Runs the specified SELECT query and then calls the specified callback with the result.
+	 * 
+	 * @param sql SQL string specifying the columns to select, tables to select from and the WHERE clause.
+	 * @param queryCallback Callback object to process the result.
+	 * 
+	 * @throws TskCoreException 
+	 */
+	public void select(final String sql, final DBAccessQueryCallback queryCallback) throws TskCoreException {
+
+		if (queryCallback == null) {
+            throw new TskCoreException("Callback is null");
+        }
+		
+		validateSQL(sql);
+		
+		CaseDbConnection connection = tskDB.getConnection();
+		tskDB.acquireSingleUserCaseReadLock();
+
+		Statement statement = null;
+		ResultSet resultSet;
+		String selectSQL = "SELECT " +  sql; // NON-NLS
+		try {
+			statement = connection.createStatement();
+			resultSet = statement.executeQuery(selectSQL);
+			queryCallback.process(resultSet);
+		} catch (SQLException ex) {
+			throw new TskCoreException("Error running SELECT query.", ex);
+		} finally {
+			closeStatement(statement);
+			connection.close();
+			tskDB.releaseSingleUserCaseReadLock();
+		}
+	}
+	
+	/**
+	 * Deletes a row in the specified table.
+	 * 
+	 * @param tableName table from which to delete the row
+	 * @param sql - SQL string specifying the condition to identify the row to delete
+	 * 
+	 * @throws TskCoreException 
+	 */
+	public void delete(final String tableName, final String sql ) throws TskCoreException {
+		validateTableName(tableName);
+		validateSQL(sql);
+
+		CaseDbConnection connection = tskDB.getConnection();
+		tskDB.acquireSingleUserCaseReadLock();
+
+		Statement statement = null;
+		String deleteSQL = "DELETE FROM " + tableName + " " + sql; // NON-NLS
+		try {
+			statement = connection.createStatement();
+			statement.executeUpdate(deleteSQL);
+		} catch (SQLException ex) {
+			throw new TskCoreException("Error deleting row from table " + tableName, ex);
+		} finally {
+			closeStatement(statement);
+			connection.close();
+			tskDB.releaseSingleUserCaseReadLock();
+		}
+	}
+
+	/**
+	 * Validates table name. 
+	 * Specifically, it ensurer the table doesn't begin with 'tsk_' 
+	 * to avoid modifications to core TSK tables
+	 * 
+	 * @param tableName
+	 * @throws TskCoreException 
+	 */
+	private void validateTableName(String tableName) throws TskCoreException {
+		
+		if (tableName.toLowerCase().startsWith("tsk_")) {
+			throw new TskCoreException("Attempt to modify a core TSK table " + tableName);
+		}
```

Comment:
I'll defer this to the CaseDbAcessManager improvement JIRA.  The list of our core table and index names should be supplied by SleuthKitCase, and ideally built dynamically - i.e. obtained from the database itself - directly or indirectly (DBMS calls or us maintining tables of names of tables/indexes)

Reply:
We can't do this dynamically by querying the database, because we will not be able to distinguish our tables from user tables...unless we enforce the use of a prefix on user tables. Alternatively, we could add a package access listing of table names to SleuthkitCase, at the cost of having to update the listing.

Reply:
Okay on deferring.